src/cocaine-app/history.py

```diff
@@ -0,0 +1,401 @@
+import logging
+
+from config import config
+from db.mongo import MongoObject
+from db.mongo.pool import Collection
+
+
+logger = logging.getLogger('mm.infrastructure')
+
+
+class GroupHistory(MongoObject):
+    """
+    Group's complete history
+
+    History objects can provide two types of changes that were recorded by mastermind:
+        - group's affiliation with a couple;
+        - node backends that were a part of group at any time.
+
+    Every state record contains timestamp and record type which represents a reason
+    why the state was changed so that process of group evolution could be easily traced.
+
+    Attributes:
+        group_id - id of a group to which history belongs;
+        couples - iterable of GroupCoupleRecord which represents the points in time
+            when group changed its affiliation with a couple;
+        nodes - iterable of GroupNodeBackendsSetRecord which represents changes
+            of underlying node backends set.
+    """
+
+    GROUP_ID = 'group_id'
+    COUPLES = 'couples'
+    NODES = 'nodes'
+
+    PRIMARY_ID_KEY = GROUP_ID
+
+    PARAMS = (GROUP_ID, COUPLES, NODES)
```

Comment:
Ah you! You were right the whole time!
https://github.com/yandex/mastermind/commit/1a928b0f200fd2a73372606396f6f1cef5c01c1e